src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.cs

```diff
@@ -0,0 +1,132 @@
+                        {
+                            // This is an invocation that contains an array as an argument
+                            // This will get caught by the first case in another cycle
+                            return;
+                        }
+
+                        argumentOperation = invocationOperation.Arguments.FirstOrDefault();
+                        if (argumentOperation is not null)
+                        {
+                            if (argumentOperation.Children.First() is not IConversionOperation conversionOperation
+                                || conversionOperation.Operand is not IArrayCreationOperation arrayCreation)
+                            {
+                                return;
+                            }
+                            arrayCreationOperation = arrayCreation;
+                        }
+                        else // An invocation, extension or regular, has an argument, unless it's a VB extension method call
+                        {
+                            // For VB extension method invocations, find a matching child
+                            arrayCreationOperation = (IArrayCreationOperation)invocationOperation.Descendants()
+                                .FirstOrDefault(x => x is IArrayCreationOperation);
+                            if (arrayCreationOperation is null)
+                            {
+                                return;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        return;
+                    }
+
+                    if (argumentOperation is not null)
+                    {
+                        ITypeSymbol originalDefinition = argumentOperation.Parameter.Type.OriginalDefinition;
+
+                        // Can't be a ReadOnlySpan, as those are already optimized
+                        if (readonlySpanType is not null && originalDefinition.Equals(readonlySpanType))
+                        {
+                            return;
+                        }
+
+                        // Check if the parameter is a function so the name can be set to null
+                        // Otherwise, the parameter name doesn't reflect the array creation as well
+                        if (functionType is not null)
+                        {
+                            isDirectlyInsideLambda = originalDefinition.Equals(functionType);
+                        }
+                    }
+
+                    // Must be literal array
+                    if (arrayCreationOperation.Initializer.ElementValues.Any(x => x is not ILiteralOperation))
+                    {
+                        return;
+                    }
+
+                    Dictionary<string, string?> properties = new()
+                    {
+                        { "paramName", isDirectlyInsideLambda ? null : argumentOperation?.Parameter?.Name }
```

Comment:
You are assuming that `Parameter` can be null here but not on line `98`, I am wondering if one place needs to be updated.